src/SpoolmanManager.cpp

```
@@ -1322,6 +1330,32 @@ bool SpoolmanManager::syncSpool(const SpoolmanSyncRequest& req, int& resolvedSpo
     resolvedSpoolmanId = -1;
     bool success = false;
 
+    // If the writer pre-registered a spool to link, consume it and patch nfc_id before syncing.
+    // This prevents auto-sync from creating a duplicate when a pre-selected spool exists.
+    // Read timestamp before exchange: setPendingLink stores time before ID, so a valid ID
+    // guarantees the timestamp is already set (no race window on the age check).
+    uint32_t linkSetAt = pendingLinkSetAt_.load();
+    int32_t linkSpoolId = pendingLinkSpoolId_.exchange(-1);
+    if (linkSpoolId > 0) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)